plugins/caddy/main.go

```diff
@@ -31,7 +32,11 @@ func init() {
 	httpcaddyfile.RegisterHandlerDirective(moduleName, parseCaddyfileHandlerDirective)
 }
 
-var staticConfig Configuration
+var (
+	staticConfig Configuration
+	appCounter = 0
+	appConfigs []*Configuration
```

Comment:
I don't think this is safe across config reloads of Caddy. You should use a `caddy.UsagePool` instead.

The reason is that when Caddy is reloading, there are two versions of the config running at the same time, concurrently, for a short period of time (old one continues to run until the new one is fully provisioned and ready to go, then old one is torn down). `UsagePool` makes sure things are properly reference counted for you, and has the appropriate mutexes.